client/src/modules/user/UserProfile/DeleteAccount.jsx

```diff
@@ -0,0 +1,172 @@
+import { deleteAccount, logoutCurrentUser } from '../model/actions';
+import './DeleteAccount.scss';
+import DeleteDialog from './DeleteDialog';
+import Dialog from 'common/components/Dialog';
+import AlertBox from 'common/components/AlertBox';
+import { MessageType } from 'common/constants/enums';
+
+class DeleteAccount extends Component {
+  state = {
+    email: '',
+    isDeleteDialogVisible: false,
+    isErrorVisible: false,
+    password: '',
+    pending: false,
+    verificationText: ''
+  };
+
+  handleDeleteAccount = async event => {
+    this.setState({ pending: true });
+    event.preventDefault();
+
+    const {
+      intl: { formatMessage }
+    } = this.props;
+    const { verificationText } = this.state;
+    const verificationString = formatMessage({ id: 'delete-form.verify-text' });
+
+    try {
+      if (verificationText === verificationString) {
+        const { email, password } = this.state;
+        const trimmedEmail = _trim(email);
+        const result = await deleteAccount(trimmedEmail, password);
+
+        if (result) {
+          this.setState({
+            isAccountDeletedDialogVisible: true,
+            isDeleteDialogVisible: false,
+            pending: false
+          });
+        }
+
+        return;
+      }
+
+      throw new Error();
+    } catch {
+      this.setState({ isErrorVisible: true, pending: false });
+    }
+  };
+
+  handleEmailChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ email: value });
+  };
+
+  handlePasswordChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ password: value });
+  };
+
+  handleVerificationText = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ verificationText: value });
+  };
+
+  showDeleteDialog = () => this.setState({ isDeleteDialogVisible: true });
+
+  hideDeleteDialog = () =>
+    this.setState({ isDeleteDialogVisible: false, isErrorVisible: false });
+
+  handleCancel = () => {
+    const { logoutCurrentUser } = this.props;
+
+    logoutCurrentUser();
+  };
+
+  render() {
+    const {
+      intl: { formatMessage }
+    } = this.props;
+    const {
+      isAccountDeletedDialogVisible,
+      isDeleteDialogVisible,
+      isErrorVisible,
+      pending
+    } = this.state;
+
+    return (
+      <Fragment>
+        {isDeleteDialogVisible && (
+          <DeleteDialog
+            error={isErrorVisible}
+            onCancel={this.hideDeleteDialog}
+            onConfirm={this.handleDeleteAccount}
+            onEmailChange={this.handleEmailChange}
+            onPasswordChange={this.handlePasswordChange}
+            onSubmit={this.handleDeleteAccount}
+            onVerificationTextChange={this.handleVerificationText}
+            pending={pending}
+          />
+        )}
+        {isAccountDeletedDialogVisible && (
+          <Dialog
+            cancelLabel="common.ok"
+            onCancel={this.handleCancel}
```

Comment:
why `cancel` is used?

Reply:
This has to be refactored on a separated task